src/o365/spo/commands/page/page-clientsidewebpart-add.ts

```diff
@@ -0,0 +1,483 @@
+				}
+				// Save the Client Side Page with updated information
+				return this._saveClientSidePage(clientSidePage, cmd, args, siteAccessToken, requestDigest);
+			})
+			.then(() => {
+				if (this.verbose) {
+					cmd.log(
+						`Client Side WebPart ${args.options.webPartId ||
+							args.options.standardWebPart} is added to page ${pageName}`
+					);
+				}
+				cb();
+			})
+			.catch((err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+	}
+
+	private _getClientSidePage(
+		pageName: string,
+		cmd: CommandInstance,
+		args: CommandArgs,
+		accessToken: string
+	): Promise<ClientSidePage> {
+		return new Promise<ClientSidePage>((resolve, reject) => {
+			if (this.verbose) {
+				cmd.log(`Retrieving information about the page...`);
+			}
+
+			const webUrl = args.options.webUrl;
+			const requestOptions: any = {
+				url: `${webUrl}/_api/web/getfilebyserverrelativeurl('${webUrl.substr(
+					webUrl.indexOf('/', 8)
+				)}/sitepages/${encodeURIComponent(pageName)}')?$expand=ListItemAllFields/ClientSideApplicationId`,
+				headers: Utils.getRequestHeaders({
+					authorization: `Bearer ${accessToken}`,
+					'content-type': 'application/json;charset=utf-8',
+					accept: 'application/json;odata=nometadata'
+				}),
+				json: true
+			};
+
+			if (this.debug) {
+				cmd.log('Executing web request...');
+				cmd.log(requestOptions);
+				cmd.log('');
+			}
+
+			request.get(requestOptions).then((res) => {
+				if (this.debug) {
+					cmd.log('Response:');
+					cmd.log(res);
+					cmd.log('');
+				}
+
+				if (res.ListItemAllFields.ClientSideApplicationId !== 'b6917cb1-93a0-4b97-a84d-7cf49975d4ec') {
+					reject(new Error(`Page ${args.options.pageName} is not a modern page.`));
+					return;
+				}
+
+				const clientSidePage: ClientSidePage = ClientSidePage.fromHtml(res.ListItemAllFields.CanvasContent1);
+
+				resolve(clientSidePage);
+			});
+		});
+	}
+
+	private _getWebPart(cmd: CommandInstance, args: CommandArgs, accessToken: string): Promise<ClientSideWebpart> {
+		return new Promise<ClientSideWebpart>((resolve, reject) => {
+			const standardWebPart = args.options.standardWebPart;
+
+			const webPartId = standardWebPart
+				? StandartWebPartUtils.getWebPartId(standardWebPart)
+				: args.options.webPartId;
+
+			if (this.debug) {
+				cmd.log(`StandardWebPart: ${standardWebPart}`);
+				cmd.log(`WebPartId: ${webPartId}`);
+			}
+
+			const webUrl = args.options.webUrl;
+			const requestOptions: any = {
+				url: `${webUrl}/_api/web/getclientsidewebparts()`,
+				headers: Utils.getRequestHeaders({
+					authorization: `Bearer ${accessToken}`,
+					'content-type': 'application/json;charset=utf-8',
+					accept: 'application/json;odata=nometadata'
+				}),
+				json: true
+			};
+
+			if (this.debug) {
+				cmd.log('Executing web request...');
+				cmd.log(requestOptions);
+				cmd.log('');
+			}
+
+			request
+				.get(requestOptions)
+				.then((res: { value: ClientSidePageComponent[] }) => {
```

Comment:
If this request fails, then the CLI will be stuck because the rejected promise isn't handled.